STM32F373: 'ADC_InitTypeDef' {aka 'struct <anonymous>'} has no member named 'ExternalTrigConvEdge'

Hi,

I am using a STM32F373. Therefore, I created a variant. Compiling this leads me to following error:

`
~/.arduino15/packages/STM32/hardware/stm32/1.6.1/cores/arduino/stm32/analog.c: In function 'adc_read_value':
~/.arduino15/packages/STM32/hardware/stm32/1.6.1/cores/arduino/stm32/analog.c:670:18: error: 'ADC_InitTypeDef' {aka 'struct <anonymous>'} has no member named 'ExternalTrigConvEdge'; did you mean 'ExternalTrigConv'?
   AdcHandle.Init.ExternalTrigConvEdge  = ADC_EXTERNALTRIGCONVEDGE_NONE; /* Parameter discarded because software trigger chosen */
                  ^~~~~~~~~~~~~~~~~~~~
                  ExternalTrigConv
exit status 1
`
To overcome this, I changed the referenced code in the analog.c as followed:

`
#if !defined(STM32F1xx) && !defined(STM32F373xC)
AdcHandle.Init.ExternalTrigConvEdge  = ADC_EXTERNALTRIGCONVEDGE_NONE; /* Parameter discarded because software trigger chosen */
#endif
`

Then, the compiling was successfull. I am not sure wether this is correct though.

Regards

Robert.

--- a/cores/arduino/stm32/analog.cpp
+++ b/cores/arduino/stm32/analog.cpp
@@ -822,7 +822,7 @@ uint16_t adc_read_value(PinName pin)
   AdcHandle.Init.NbrOfDiscConversion   = 0;                             /* Parameter discarded because sequencer is disabled */
 #endif
   AdcHandle.Init.ExternalTrigConv      = ADC_SOFTWARE_START;            /* Software start to trig the 1st conversion manually, without external event */
-#if !defined(STM32F1xx)
+#if !defined(STM32F1xx) && !defined(STM32F373xC) && !defined(STM32F378xx)
   AdcHandle.Init.ExternalTrigConvEdge  = ADC_EXTERNALTRIGCONVEDGE_NONE; /* Parameter discarded because software trigger chosen */
 #endif
 #if !defined(STM32F1xx) && !defined(STM32H7xx) && \

